.github/workflows/pr-project-label.yml

```
@@ -22,3 +22,86 @@ jobs:
+            run: |
+              if [[ ! "$GITHUB_REF" =~ refs/pull/.+/merge && ! "$GITHUB_REF" =~ refs/heads/.+ ]]; then
+                echo "Error: Unexpected ref format ($GITHUB_REF). Exiting."
+                exit 1
+              fi
+              
+          - name: Detect changelog types from content
+            env:
+              GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+              PR_NUMBER: ${{ github.event.pull_request.number }}
+            run: |
+              declare -A LABEL_MAP=(
+                ["fix"]="type: fix"
+                ["add"]="type: feature"
+                ["update"]="type: enhancement"
+                ["tweak"]="type: enhancement"
+                ["enhancement"]="type: enhancement"
+                ["performance"]="type: performance"
+                ["dev"]="type: infrastructure"
+              )
+    
+              APPLIED_LABELS=()
+    
+              # Get list of changed files in the PR and filter to changelog files
+              CHANGED_FILES=$(gh pr view "$PR_NUMBER" --json files -q '.files[].path' | grep '^plugins/woocommerce/changelog/' || true)
+    
+              echo "Changed files detected:"
+              echo "$CHANGED_FILES"
+    
+              # Loop through only changed changelog files
+              for FILE in $CHANGED_FILES; do
+                FULL_PATH="$GITHUB_WORKSPACE/$FILE"
+    
+                if [[ -f "$FULL_PATH" ]]; then
+                  echo "Processing file: $FULL_PATH"
+    
+                  # Extract the type from file content
+                  TYPE=$(grep -iE '^Type: (fix|add|update|tweak|enhancement|performance|dev)' "$FULL_PATH" | awk -F': ' '{print tolower($2)}' | head -n 1)
+                  echo "Type: $TYPE"
+
+                  # Validate extracted type
+                  if [[ ! "${LABEL_MAP[$TYPE]}" ]]; then
+                    echo "Invalid changelog type found in $FULL_PATH: '$TYPE'. Skipping."
+                    continue
```

Should it not fail if the type was not found? It could signal a change in the changelog types that we want to include in here.